app/controllers/api/v1/books_controller.rb

```diff
@@ -13,6 +13,11 @@ def show
         book = Book.includes(:rents).find(params[:id])
         render json: book, serializer: Books::ShowSerializer
       end
+
+      def search
+        book = OpenLibraryService.new.book_info(params[:search])
+        render json: book
```

Comment:
If you make the changes below, and add a rescue_from that exception you can control the error code. If not, it will always respond with 200 status, even if no book was found.